### PR TITLE
common/lru: unit-test(not complete)

### PR DESCRIPTION
**Pull request by ShadowErii**
Go

### CODE REVIEW

**Comment by ucwong** [comment]
https://github.com/hashicorp/golang-lru/blob/master/lru_test.go

Good, I think this is a good example for you

**Comment by ShadowErii** [comment]
> https://github.com/hashicorp/golang-lru/blob/master/lru_test.go
> https://github.com/hashicorp/golang-lru/blob/master/simplelru/lru_test.go
> 
> Good, I think this is a good example for you

File: common/lru/lru_test.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+	l := New(2)
+	l.Add(1, 1, 1)
+	l.Add(1, 1, 1)
+	if l.Len() > 1 {
+		t.Fatalf("the same cache should not add again")
+	}
+
+	l.Add(1, 1, 1)
+	l.Add(2, 2, 2)
+}
```
Comment (ucwong) [inline common/lru/lru_test.go:99]: Add expected result check for unit test, for example 
https://github.com/hashicorp/golang-lru/blob/master/simplelru/lru_test.go#L111
```
// Test that Add returns true/false if an eviction occurred
func TestLRU_Add(t *testing.T) {
	evictCounter := 0
	onEvicted := func(k interface{}, v interface{}) {
		evictCounter++
	}

	l, err := NewLRU(1, onEvicted)
	if err != nil {
		t.Fatalf("err: %v", err)
	}

	if l.Add(1, 1) == true || evictCounter != 0 {
		t.Errorf("should not have an eviction")
	}
	if l.Add(2, 2) == false || evictCounter != 1 {
		t.Errorf("should have an eviction")
	}
}
```

File: common/lru/lru_test.go [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+	if _, ok := l.Get(200); ok {
+		t.Fatalf("should contain nothing")
+	}
+}
+
+func TestLRU_RemoveOldest(t *testing.T) {
+	l := New(128)
+	for i := 0; i < 256; i++ {
+		l.Add(i, i, 1)
+	}
```
Comment (ucwong) [inline common/lru/lru_test.go:84]: https://github.com/hashicorp/golang-lru/blob/master/simplelru/lru_test.go#L78

File: common/lru/lru_test.go (hunk 1 above)
Comment (DhunterAO) [reply common/lru/lru_test.go:99]: If there is no return values for a function, you can check whether the state after function call is expected or not.

Comment (ShadowErii) [reply common/lru/lru_test.go:99]: got it

**Comment by ucwong** [comment]
You can change its status when it is ready

**Pull request merged**